src/language/pattern-types.md

```diff
@@ -0,0 +1,333 @@
+---
+title: Pattern types
+description: Pattern type reference in Dart.
+---
+
```

Comment:
Maybe a one-sentence intro here that links back to the other patterns page, like:

```
Here are all the kinds of [patterns][] that can be used.
```